### PR TITLE
 [FIX] point_of_sale: UI print sale details only for current config

### PR DESCRIPTION
**Pull request by switch87**
PURPOSE

If the printer icon inside a PoS session is clicked, a ticket will be printed with a report of all sales of today for the curent PoS Config.

SPECIFICATIONS

- At the moment If clicked on the icon a report will be printed for all sales of the day, including all other PoS configs. This is not wanted behaviour.
    original pr with fix: https://github.com/odoo/odoo/pull/32789
- Only managers should be able to print this report

### CODE REVIEW

File: addons/point_of_sale/models/pos_order.py [hunk 1]
```diff
@@ -1175,41 +1175,48 @@ class ReportSaleDetails(models.AbstractModel):
...
+
+        domain = [
+            ('date_order', '>=', fields.Datetime.to_string(date_start)),
+            ('date_order', '<=', fields.Datetime.to_string(date_stop)),
+            ('state', 'in', ['paid','invoiced','done'])]
 
-        # avoid a date_stop smaller than date_start
-        date_stop = max(date_stop, date_start)
+        if config_ids:
+            domain.append(('config_id', 'in', config_ids))
```
Comment (pimodoo) [inline addons/point_of_sale/models/pos_order.py]: Use "expression.AND" instead of append

File: addons/point_of_sale/models/pos_order.py [hunk 2]
```diff
@@ -1175,41 +1175,48 @@ class ReportSaleDetails(models.AbstractModel):
...
 
-        date_start = fields.Datetime.to_string(date_start)
-        date_stop = fields.Datetime.to_string(date_stop)
+        if (session_ids):
```
Comment (pimodoo) [inline addons/point_of_sale/models/pos_order.py]: I've just talked with Laura, and she said that if session_ids are given, we shouldn't take into account the dates (and the config). But the state in ['paid', 'invoiced', 'done'] still needed in all cases.

The arguments date_start, date_stop and config_ids are only used when we use the backend wizard to generate the report.

File: addons/point_of_sale/static/src/js/devices.js [hunk 3]
```diff
@@ -480,11 +480,24 @@ var ProxyDevice  = core.Class.extend(mixins.PropertiesMixin,{
...
     },
 
-    print_sale_details: function() {
+    /** Print an overview of todays sales.
+     *
+     * By default this will print all sales of the day for current PoS config.
+     * 
+     * @param {boolean} current_session: If set, only sales for the current session will be printed.
+     */
+    print_sale_details: function(current_session) {
```
Comment (pimodoo) [inline addons/point_of_sale/static/src/js/devices.js]: no need to set argument current_session as called only once and you manually set true

File: addons/point_of_sale/static/src/js/devices.js [hunk 4]
```diff
@@ -480,11 +480,24 @@ var ProxyDevice  = core.Class.extend(mixins.PropertiesMixin,{
...
         var self = this;
+        var config_ids = [this.pos.config.id];
```
Comment (pimodoo) [inline addons/point_of_sale/static/src/js/devices.js]: not needed as session_id is in the config.

File: addons/point_of_sale/static/src/js/models.js [hunk 5]
```diff
@@ -175,7 +175,7 @@ exports.PosModel = Backbone.Model.extend({
         loaded: function(self,users){
             self.user = users[0];
             self.user.role = 'manager';
-            self.employee.name = self.user.name;
+            self.employee = self.user;
```
Comment (pimodoo) [inline addons/point_of_sale/static/src/js/models.js]: Why do you change it here? If really needed maybe do a separate commit

Comment (switch87) [reply addons/point_of_sale/static/src/js/models.js]: Probably a leftover from before spec change, like the always true argument earlier. Removing it :wink:

**Comment by pimodoo** [comment]
robodoo r+
